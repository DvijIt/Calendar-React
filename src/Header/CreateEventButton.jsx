import React, { PureComponent } from 'react'

class CreateButton extends PureComponent {

  render() {
    return (
      <div className="header__create-event">
        <button
          className="waves-effect waves-teal z-depth-2 white black-text create-btn"
          onClick={this.props.openPopup}
          data-option="buttonCreateEvent">
          Создать
        </button>
      </div>

    )
  }
}

export default CreateButton