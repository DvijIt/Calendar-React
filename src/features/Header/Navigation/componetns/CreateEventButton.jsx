import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class CreateButton extends PureComponent {
  render() {
    const { openPopup } = this.props;
    return (
      <div className="header__create-event">
        <button
          type="button"
          className="waves-effect waves-teal z-depth-2 white black-text create-btn"
          onClick={openPopup}
          data-option="buttonCreateEvent"
        >
          Создать
        </button>
      </div>

    );
  }
}

CreateButton.propTypes = {
  openPopup: PropTypes.func,
};

CreateButton.defaultProps = {
  openPopup: PropTypes.func,
};

export default CreateButton;
