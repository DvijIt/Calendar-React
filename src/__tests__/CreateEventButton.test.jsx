import React from 'react'
import { shallow } from 'enzyme'
import CreateEventButton from '../Header/CreateEventButton'

describe('<CreateEventButton />', () => {
  it('should render create button', () => {
    const wrappedComponent = shallow(<CreateEventButton />)
    expect(wrappedComponent).toMatchSnapshot()
  })

  it('should open modal on click button', () => {
    const mockOpenPopup = jest.fn()
    const wrappedComponent = shallow(<CreateEventButton openPopup={mockOpenPopup} />)
    wrappedComponent.find('.create-btn').simulate('click')
    expect(mockOpenPopup).toBeCalled()
  });
})