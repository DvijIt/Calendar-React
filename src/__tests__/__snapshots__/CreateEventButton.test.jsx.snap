// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateEventButton /> should render create button 1`] = `
<div
  className="header__create-event"
>
  <button
    className="waves-effect waves-teal z-depth-2 white black-text create-btn"
    data-option="buttonCreateEvent"
  >
    Создать
  </button>
</div>
`;
